addons/binding/org.openhab.binding.xiaomivacuum/README.md

```diff
@@ -0,0 +1,71 @@
+# Xiaomi Robot Vacuum Binding
+
+This Binding is used to control a Xiaomi Robot Vacuum.
+
+## Supported Things
+
+This Binding supports Xiaomi Robot Vacuum devices.
+
+## Discovery
+
+The binding needs a token from the Xiaomi Robot Vacuum in order to be able to control it.
+
+Below firmware version firmware version 3.3.9_003073:
+In order to fetch the token, reset the robot vacuum, connect to its the network its announcing (rockrobo-XXXX) and run the discovery again. After the token is retrieved you can connect the vacuum to your phone again.
+Once connected to your phone & the regular wifi network, run discovery once more to retrieve the new ipaddress.
+
+Firmware version 3.3.9_003073 & higher
+Newer firmware change the token upon inclusion in the mihome app. The token needs to be retrieved from the application database (find the mio2db file with and read it sqlite). The easiest way to do is by using [MiToolkit](https://github.com/ultrara1n/MiToolkit/releases)
+
```

Comment:
Yes, you can do a un-encrypted iTunes-Backup and use a tool like this to view the files in it: [](http://www.imactools.com/iphonebackupviewer/)
Then search in "RAW, com.xiaomi.home," for "USERID_mihome.sqlite" and look for the 32-digit-token.